cp1404practicals/prac_09/cleanup_files.py

```diff
@@ -0,0 +1,48 @@
+"""
+CP1404/CP5632 Practical
+Demos of various os module examples
+"""
+import os
+import re
+
+
+def main():
+    """Demo os module functions."""
+    print("Starting directory is: {}".format(os.getcwd()))
+
+    # Change to desired directory
+    os.chdir('Lyrics/Christmas')
+
+    # Print a list of all files in current directory
+    print("Files in {}:\n{}\n".format(os.getcwd(), os.listdir('.')))
+
+    # Loop through each file in the (current) directory
+    for filename in os.listdir('.'):
+        # Ignore directories, just process files
+        if os.path.isdir(filename):
+            continue
+
+        if ' ' in filename:
+            # replace any space with underscore
+            new_name = get_fixed_filename(filename)
+
+        elif ' ' not in filename:
+            # insert underscore between any lower and uppercase characters
+            temp_name = re.sub(r'((?<=[a-z])[A-Z]|(?<!\A)[A-Z](?=[a-z]))', r' \1', filename)
+            new_name = temp_name.replace(" ", "_").replace(".TXT", ".txt")
```

Comment:
Possibly for future use, could replace the above looping through each letter (where you check if the next one is upper case, etc) mainly because we haven't yet been taught the `temp_name = re.sub(r'...)` part but also because a name like 'harkThe HeraldAngelsSing' wouldn't be completely fixed (your if statement means that it would change the space but not add the underscores between the other words. 
Also, the elif is not required as if `' ' in filename` returns false, then `' ' not in filename' will be true.